File: examples/ApiAppUpdateExample.ts
import * as fs from 'fs';
import api from "@dropbox/sign"
import models from "@dropbox/sign"

const apiCaller = new api.ApiAppApi();
apiCaller.username = "YOUR_API_KEY";
// apiCaller.accessToken = "YOUR_ACCESS_TOKEN";

const oauth = new models.SubOAuth();
oauth.callbackUrl = "https://example.com/oauth";
oauth.scopes = [
    models.SubOAuth.ScopesEnum.BasicAccountInfo,
    models.SubOAuth.ScopesEnum.RequestSignature,
];

const whiteLabelingOptions = new models.SubWhiteLabelingOptions();
whiteLabelingOptions.primaryButtonColor = "#00b3e6";
whiteLabelingOptions.primaryButtonTextColor = "#ffffff";

const apiAppUpdateRequest = new models.ApiAppUpdateRequest();
apiAppUpdateRequest.callbackUrl = "https://example.com/dropboxsign";
apiAppUpdateRequest.name = "New Name";
apiAppUpdateRequest.domains = [
    "example.com",
];
apiAppUpdateRequest.customLogoFile = fs.createReadStream("CustomLogoFile.png");
apiAppUpdateRequest.oauth = oauth;
apiAppUpdateRequest.whiteLabelingOptions = whiteLabelingOptions;

const clientId = "0dd3b823a682527788c4e40cb7b6f7e9";

apiCaller.apiAppUpdate(
    clientId,
    apiAppUpdateRequest,
).then(response => {
  console.log(response.body);
}).catch(error => {
  console.log("Exception when calling ApiApp#apiAppUpdate:");
  console.log(error.body);
});

File: examples/ApiAppUpdateExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Dropbox.Sign.Api;
using Dropbox.Sign.Client;
using Dropbox.Sign.Model;

namespace Dropbox.SignSandbox;

public class ApiAppUpdateExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.Username = "YOUR_API_KEY";
        // config.AccessToken = "YOUR_ACCESS_TOKEN";

        var oauth = new SubOAuth(
            callbackUrl: "https://example.com/oauth",
            scopes: [
                SubOAuth.ScopesEnum.BasicAccountInfo,
                SubOAuth.ScopesEnum.RequestSignature,
            ]
        );

        var whiteLabelingOptions = new SubWhiteLabelingOptions(
            primaryButtonColor: "#00b3e6",
            primaryButtonTextColor: "#ffffff"
        );

        var apiAppUpdateRequest = new ApiAppUpdateRequest(
            callbackUrl: "https://example.com/dropboxsign",
            name: "New Name",
            domains: [
                "example.com",
            ],
            customLogoFile: new FileStream(
                path: "CustomLogoFile.png",
                mode: FileMode.Open
            ),
            oauth: oauth,
            whiteLabelingOptions: whiteLabelingOptions
        );

        try
        {
            var response = new ApiAppApi(config).ApiAppUpdate(
                clientId: "0dd3b823a682527788c4e40cb7b6f7e9",
                apiAppUpdateRequest: apiAppUpdateRequest
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling ApiAppApi#ApiAppUpdate: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}
